lib/Core/BuildEngine.cpp

```diff
@@ -965,6 +965,8 @@ class BuildEngineImpl {
     
     // Find the cycle by searching from the entry node.
     struct WorkItem {
+      WorkItem(Rule * node) { this->node = node; }
+
```

Comment:
The following code works with Clang but not MSVC. I can make a connect bug report if you think its an MSVC bug. I've been trying to make bug reports for each difference between the two compilers whilst porting Swift to MSVC:
```
int main()
{
  struct WorkItem {
    int node;
    unsigned predecessorIndex = 0;
  };  

  WorkItem{ 0 };

  return 0;
}
```